services/QuillLMS/client/app/bundles/Connect/components/activityHealth/activityHealth.tsx

```diff
@@ -0,0 +1,417 @@
+import * as React from 'react'
+import ReactTable from 'react-table'
+import { matchSorter } from 'match-sorter';
+import 'react-table/react-table.css';
+import request from 'request'
+import _ from 'underscore'
+import stripHtml from "string-strip-html"
+import { CSVLink } from 'react-csv'
+import { connect } from 'react-redux';
+
+import PromptHealth from './promptHealth'
+import NumberFilterInput from './numberFilterInput'
+
+import LoadingSpinner from '../shared/loading_indicator.jsx'
+import { sort, sortByList } from '../../../../modules/sortingMethods.js'
+import { FlagDropdown } from '../../../Shared/index'
+import { filterNumbers } from '../../../../modules/filteringMethods.js'
+import actions from '../../actions/activityHealth'
+
+const CONNECT_TOOL = "connect"
+const ACTIVITY_HEALTHS_URL = `${process.env.DEFAULT_URL}/api/v1/activities/activities_health.json`
+const ALL_FLAGS = "All Flags"
+const NO_DATA_FOUND_MESSAGE = "Activity Health data could not be found. Refresh to try again, or contact the engineering team."
+
+interface ActivityHealthProps {
+  dispatch: Function;
+  activityHealth: {
+    flag: string;
+  }
+}
+
+interface ActivityHealthState {
+  loadingTableData: boolean;
+  flag: string;
+  fetchedData: Array<Object>;
+  promptSearchInput: string;
+  dataToDownload: Array<Object>;
+}
+
+class ActivityHealth extends React.Component<ActivityHealthProps, ActivityHealthState> {
```

Comment:
NBD for this PR, but as a note for going forward, we have been trying to write new React components as functional components, using the React Hook syntax instead of declaring state.